compiler/z/codegen/FPTreeEvaluator.cpp

```diff
@@ -1165,58 +1165,24 @@ OMR::Z::TreeEvaluator::dnegEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    return unaryEvaluator(node, cg, TR::InstOpCode::LCDBR);
    }
 
-/**
- * Type-coerce int to float
- */
+// type coercion helper for fbits2i and dbits2l
 TR::Register *
-OMR::Z::TreeEvaluator::ibits2fEvaluator(TR::Node * node, TR::CodeGenerator * cg)
-   {
-   //Disable FPE Version For Now Due To Performance Reasons
-   bool disabled = true;
-   TR::Node * firstChild = node->getFirstChild();
-   TR::Register * targetReg = cg->allocateRegister(TR_FPR);
-   TR::Register * sourceReg;
-   sourceReg = cg->evaluate(firstChild);
-   if (TR::Compiler->target.cpu.getS390SupportsFPE() && !disabled)
-      {
-      TR::Register *tempreg;
-      tempreg = cg->allocate64bitRegister();
-      generateRSInstruction(cg, TR::InstOpCode::SLLG, node, tempreg, sourceReg, 32);
-      generateRRInstruction(cg, TR::InstOpCode::LDGR, node, targetReg, tempreg);
-      cg->stopUsingRegister(tempreg);
-      }
-   else
-      {
-      TR::SymbolReference * i2fSR = cg->allocateLocalTemp(TR::Int32);
-      TR::MemoryReference * tempMR = generateS390MemoryReference(node, i2fSR, cg);
-      TR::MemoryReference * tempMR1 = generateS390MemoryReference(node, i2fSR, cg);
-      generateRXInstruction(cg, TR::InstOpCode::ST, node, sourceReg, tempMR);
-      generateRXInstruction(cg, TR::InstOpCode::LE, node, targetReg, tempMR1);
-      }
-   node->setRegister(targetReg);
-   cg->decReferenceCount(firstChild);
-   return targetReg;
-   }
-
-/**
- * Type-coerce float to int
- */
-TR::Register *
-OMR::Z::TreeEvaluator::fbits2iEvaluator(TR::Node * node, TR::CodeGenerator * cg)
+FPtoIntBitsTypeCoercionHelper(TR::Node * node, TR::CodeGenerator * cg)
    {
    TR::Register * targetReg;
    TR::Register * sourceReg;
```

Comment:
Can we initialize these values so if they're accidentally used we won't have undefined behavior.

Reply:
3ba28f5190d6507418b1678a02e6ee7a359bc67e